crates/net/common/src/metered_sender.rs

```diff
@@ -0,0 +1,64 @@
+//! Support for metering senders. Facilitates debugging by exposing metrics for number of messages
+//! sent, number of errors, etc.
+
+use metrics::Counter;
+use reth_metrics_derive::Metrics;
+use tokio::sync::mpsc::{
+    error::{SendError, TrySendError},
+    Sender,
+};
+
+/// Network throughput metrics
+#[derive(Metrics)]
+#[metrics(scope = "metered_sender")]
+struct MeteredSenderMetrics {
+    /// Number of messages sent
+    messages_sent: Counter,
+    /// Number of errors when sending messages
+    send_errors: Counter,
+}
```

Comment:
the problem with this is now, that we can not differentiate between two different kinds of channels.

for example, if we'd use this for channel of type `A` and channel of type `B` then we don't know which channel incremented the counters.

so I think we should use a `&'static str` approach here, instead of deriving the identifiers.

Reply:
Exactly, ref https://github.com/paradigmxyz/reth/pull/726#discussion_r1064130656

Reply:
Gotcha, just won't use metric-derive then

Reply:
gonna take a stab at handling #782 first to address this